docs/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,195 @@
+# GoCryptoTrader Coding Guidelines
+
+This document outlines the coding, formatting, and testing standards for implementing or refactoring exchange API integrations or any related functionality within the codebase. These practices ensure consistency, maintainability, and performance throughout the project.
+
+## General Standards
+
+- Code must adhere to the official Go [formatting](https://golang.org/doc/effective_go.html#formatting) guidelines (i.e. uses [gofmt](https://golang.org/cmd/gofmt/)).
+- Code must adhere to these [Effective Go](https://go.dev/doc/effective_go) guidelines.
+- Code must also follow these [Go Style](https://google.github.io/styleguide/go/) guidelines.
+
+## Exchange Implementation Guidelines
+
+Refer to the [ADD_NEW_EXCHANGE.md](/docs/ADD_NEW_EXCHANGE.md) document for comprehensive steps on integrating a new exchange.
+
+### Endpoint Organisation
+
+- Implement API endpoints in the order they are presented in the API documentation to maintain alignment with the source.
+- Group related endpoints into files that follow the documented API structure.
+- Inline endpoint paths directly in the method implementation. Avoid defining them as constants elsewhere.
+- Export exchange types, functions and methods by default (e.g. `func (e *Exchange) GetOrderBook(...)`) so that GoCryptoTrader can be consumed as both a standalone library and interfaced via the engine package.
+
+### Type Usage
+
+- Use the most appropriate native Go types for struct fields:
+  - If the API returns numbers as strings, use float64 with the `json:",string"` tag.
+  - For timestamps, use `time.Time` if Go's JSON unmarshalling supports the format directly.
+- If native Go types are not supported directly, use the following built-in types:
+  - `types.Time` for Unix timestamps that require custom unmarshalling.
+  - `types.Number` for numerical float values where an exchange API may return either a `string` or `float64` value.
+- Always use full and descriptive field names for clarity and consistency. Avoid short API-provided aliases unless compatibility requires it.
+- Default to `uint64` for exchange API parameters and structs for integers where appropriate.
+  - Avoid `int` (size varies by architecture) or `int64` (allows negatives where they don't make sense).
+  - Aligns well with `strconv.FormatUint`.
+
+### TestMain usage
+
+- TestMain must avoid API calls to keep the test footprint as minimal as possible unless there's a good case with commentary as to why it was done.
```

Comment:
```suggestion
- TestMain must avoid API calls, so that individual unit tests can run quickly. Use sync.Once or similar patterns to bootstrap common data without burdening all unit tests with the same overhaed. See UpdatePairsOnce for an example of this.
```